### PR TITLE
Improve test case failure/skip states and their output (#4116)

### PR DESCRIPTION
**Pull request by redstrate**
### This changes visible test output

The following changes are proposed:

- Change the test output message when encountering an error exit code.
- Check for skipped tests and mark them as skipped.

## The purpose of this change

Currently if a test fails, it's not immediately clear why that is because we give the exit code - and not the output from ctest. That is now fixed and the ctest output is shown which greatly improves the usefulness of the Test Results pane and other places where this output is shown like Code Lens.

Skipped test cases were also not actually marked as skipped, and that is now fixed as well.

## Other Notes/Information

Comparison screenshots to show the improvement:

### Failed Tests

| Before | After |
| --- | --- |
| Screenshot_20241015_125716 | Screenshot_20241015_125209 |

### Skipped Tests

| Before | After |
| --- | --- |
| Screenshot_20241015_125700 | Screenshot_20241015_125759 |

Unfortunately this google test example I used doesn't set test source location, but if it did then code lens would be affected as well like in this outdated screenshot from the issue:

image

### CODE REVIEW

**Comment by redstrate** [comment]
@microsoft-github-policy-service agree company="KDAB"

**Review by gcampbell-msft — CHANGES_REQUESTED** [review]
@redstrate This looks great! 

If you could please make a CHANGELOG entry, then we can approve and accept this PR! Thanks.

**Comment by redstrate** [comment]
> @redstrate This looks great!
> 
> If you could please make a CHANGELOG entry, then we can approve and accept this PR! Thanks.

Done!

**Review by gcampbell-msft — APPROVED** [review]

File: src/ctest.ts [hunk 1]
```diff
@@ -550,15 +550,19 @@ export class CTestDriver implements vscode.Disposable {
...
+                    new vscode.TestMessage(localize('test.failed.with.exit.code', '{0}\nTest {1} failed with exit code {2}.', output, testName, exitCode)),
                     failureDuration
                 );
             } else if (completionStatus !== undefined) {
-                this.ctestErrored(
-                    test,
-                    run,
-                    new vscode.TestMessage(localize('test.failed.with.completion.status', 'Test {0} failed with completion status "{1}".', testName, completionStatus))
-                );
+                if (completionStatus == "SKIP_REGULAR_EXPRESSION_MATCHED") {
```
Comment (gcampbell-msft) [inline src/ctest.ts:557]: Needs to be `===` instead of `==`. Thanks.

**Review by gcampbell-msft — CHANGES_REQUESTED** [review]
See other comments.

**Review by gcampbell-msft — APPROVED** [review]

**Pull request merged**